x-pack/plugins/maps/public/shared/layers/sources/kibana_tilemap_source/kibana_tilemap_source.js

```diff
@@ -51,12 +52,19 @@ export class KibanaTilemapSource extends  AbstractTMSSource {
     });
   }
 
-
   async getUrlTemplate() {
-    return this._descriptor.url;
+    const tilemap = await getKibanaTileMap();
+    if (!tilemap || !tilemap.url) {
+      throw new Error(`Unable to find map.tilemap.url configuration in the kibana.yml`);
+    }
+    return tilemap.url;
   }
 
   async getDisplayName() {
-    return await this.getUrlTemplate();
+    try {
+      return await this.getUrlTemplate();
+    } catch (e) {
+      return '';
```

Comment:
This provide a blank layer name in the TOC. Maybe 'maps.tilemap.url' is better?

Reply:
it doesn't. when we don't have a layer-name (e.g. empty string), it default to the layer-id.